components/nexus-core/src/main/java/org/sonatype/nexus/proxy/item/ChecksumReconciler.java

```diff
@@ -114,7 +129,46 @@ void reconcileItemChecksum(final Repository repo, final StorageFileItem item) th
         finally {
           uid.getLock().unlock();
         }
+
+        final String itemPath = item.getPath();
+        if (!itemPath.endsWith(".sha1") && !itemPath.endsWith(".md5")) {
```

Comment:
Actually it might be ok moving the check up, since if the attributes of the main item are affected then the attributes of checksums would also be affected - and the act of storing the new value would refresh their attributes.